Sources/NonEmpty/NonEmpty+Dictionary.swift

```diff
@@ -4,7 +4,7 @@ public protocol _DictionaryProtocol: Collection where Element == (key: Key, valu
   associatedtype Key: Hashable
   associatedtype Value
   var keys: Dictionary<Key, Value>.Keys { get }
-  subscript(key: Key) -> Value? { get }
+  subscript(key: Key) -> Value? { get set }
```

Comment:
`NonEmpty` itself doesn't conform to this protocol, but we need to know that this is settable to mutate the underlying dictionary in the convenience initializer below.